src/main/webapp/app/entities/programming-exercise/programming-exercise-detail.component.html

```diff
@@ -78,6 +78,13 @@ <h2><span jhiTranslate="arTeMiSApp.programmingExercise.detail.title">Programming
                     class="btn btn-primary">
                 <fa-icon [icon]="'pencil-alt'"></fa-icon>&nbsp;<span jhiTranslate="entity.action.edit"> Edit</span>
             </button>
+
+            <button type="button"
+                    (click)="generateStructureOracle()"
+                    class="btn btn-secondary">
+                <span>Generate Structure Oracle</span>
```

Comment:
The translation is missing. Please properly use jhiTranslate and provide a German translation in the corresponding json file as well

Reply:
Added translations! :)